### ISSUE TITLE
Hide and de-prioritise Links Manager (unless site has existing Links)

### ISSUE THREAD

**Issue by donalirl** [Automattic/wp-calypso#48177]
At the moment we are presenting "Links" as the sixth option in the nav (before Pages!) Such a high position communicates to users that "Links" is a very important feature that needs to be used frequently. However, Links is a niche feature used for Blogrolls and not a popular part of the modern WP.com website building experience. I am proposing that we :

* do not include "Links" in the new navigation or
* de-priortize it in the design by moving it much further down

#### Steps to reproduce

1. Starting at URL: WordPress.com
2. Click on "Links"

#### Screenshot / Video

<img width="699" alt="Links" src="https://user-images.githubusercontent.com/30727666/101654092-385b2400-3a94-11eb-95bd-02df0df09784.PNG">

#### Context / Source

#manual-testing #perspective

**Comment by cpapazoglou** [comment]
Thanks for reporting this!

I can see your point, I am not sure that we want to deviate from how `/wp-admin` orders the menus. It seems like a suitable core issue though.

@obenland, any thoughts here?

**Comment by obenland** [comment]
Yeah, I've been surprised by seeing Links in there at all and thought it was due to my account being so old. It's definitely worth investigating if we're now showing it to users that shouldn't see it and what we can do to avoid that.

**Comment by cpapazoglou** [comment]
Searching for `link-manager.php` in the code it seems:
1) It is removed since 3.5 https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/bookmark.php#L333-L351
2) Maybe we should do that check https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/upgrade.php#L1877 before showing the menu?
3) Calypsoify used to remove that menu

**Comment by mmtr** [comment]
I see the "Links" option has been there quite a while on WP Admin for Simple sites. This is the current menu when the Nav Unification is disabled:

<img width="179" alt="Screen Shot 2021-01-22 at 17 59 56" src="https://user-images.githubusercontent.com/1233880/105521032-afaee500-5cdb-11eb-9187-f8fc7a80abf0.png">

Seems to be the intended behavior on WordPress Core: https://core.trac.wordpress.org/browser/trunk/src/wp-admin/menu.php?rev=49837#L80

The deprecation mentioned by @cpapazoglou seems is not actually a deprecation, but a message shown to users accessing the link manager without the required `manage_links` capability (or if there is a third-party plugin that is artificially overriding the `manage_links` capability).

Do we want to diverge from the Core behavior? If so, I'd vote from moving the menu option down, so legit users using this feature are not affected.

**Comment by getdave** [comment]
Nav Unification is intended to mirror WP Core navigation. The more we diverge from this then the more maintenance we incur and the more likely things go wrong.

The [WP.org Codex says of `Links Manager`](https://codex.wordpress.org/Links_Manager):

> As of Version 3.5, the Links Manager and blogroll are hidden for new installs and any existing WordPress installs that do not have any links. 

...and:

> If you are upgrading from a previous version of WordPress with any active links, the Links Manager will continue to function as normal. 

Can we 

1. Hide `Links` by default.
2. Show `Links` if the user has active links.
3. In either case, depriortise `Links` by moving down below `Feedback` on both Calypso and WPAdmin.

This should solve all the above use cases.

**Comment by getdave** [comment]
This is marked as `In Progress` but no PR is referenced. I'm moving out of priority list for now as there are more pressing items. 

If someone is working on it then please let me know.

**Comment by mmtr** [comment]
> This is marked as In Progress but no PR is referenced

I wasn't aware of that limitation. I'm used to move issues to "In progress" as soon as I start investigating them, when a PR is still not available, so others don't waste their time doing the same investigation in parallel.

**Comment by keoshi** [comment]
Questions on this item:

- Given that all suboptions exist only in wp-admin, shouldn't the Link item have an icon indicating it's an external link?
- Or are we explicitly not adding the icon _because_ this option contains a flyout menu?

![image](https://user-images.githubusercontent.com/390760/109533727-74f65480-7ab2-11eb-9d57-c806119aece3.png)

Still, when clicking “Link” I'm redirected to wp-admin, so we should make that more obvious from the get go IMO.

cc @davemart-in

**Comment by davemart-in** [comment]
Flagging this issue for @sfougnier to weigh in.

**Comment by ghost** [comment]
Thanks for the ping @davemart-in!

>- Hide Links by default.
>- Show Links if the user has active links.
>- In either case, deprioritize Links by moving down below Feedback on both Calypso and WPAdmin.

This seems ideal.

>Given that all suboptions exist only in wp-admin, shouldn't the Link item have an icon indicating it's an external link?

While I understand that _technically_ this is redirecting to wp-admin from Calypso, this won't necessarily be noticeable to users. For the sake of making things feel as unified as possible, I would say we should not add external link icons.

**Comment by keoshi** [comment]
@sfougnier I feel like maybe I should provide more context on this particular point of view.

As far I can tell this doesn't apply to Simple or Atomic sites, but in case you're going through this flow using a **Jetpack** site then you'll definitely notice the change for one or more reasons:

- All the suboptions lead you to wp-admin (as pictured in the image above).
- Most likely, the user color scheme on wp-admin will differ from whatever is available on WordPress.com.
- The sidebar width and styling changes dramatically.
- The sidebar will most likely show many more items that weren't displayed before (plugins, CPTs, etc.).
- Last but not least, the content pane is completely different from patterns used in Calypso.

This makes for a pretty jarring experience, mainly because the expectation wasn't set as to what was going to happen. If other links with the same behavior show the icon, I don't see why this one shouldn't be consistent.

I recorded a quick video so you can see exactly what happens, but I'd also encourage anyone looking at this to [spin up a Jetpack site](https://jurassic.ninja/create/) to test the flow for themselves.

https://user-images.githubusercontent.com/390760/109569702-bdc50200-7ae0-11eb-836a-4b9c238d3701.mov

This is just a suggestion but if we end up _not_ adding the icon, I think we should document these exceptions.

**Comment by mmtr** [comment]
@keoshi is correct, @sfougnier. The external icons are only used on Jetpack sites, since we don't have a unified experience there (see p1HpG7-bjx-p2).

I agree that the best solution is to hide/deprioritize the "Links" menu items on Simple/Atomic sites as noted in https://github.com/Automattic/wp-calypso/issues/48177#issuecomment-766857120, and completely hide it on Jetpack sites.

**Comment by mmtr** [comment]
> completely hide it on Jetpack sites.

FWIW, this is already the case as of https://github.com/Automattic/jetpack/pull/18921, which is included in Jetpack 9.5 (to be released today).

**Comment by ghost** [comment]
Thanks for the context @keoshi & @mmtr. The icon aspect seems like potentially a separate issue since there are other nav items with similar behavior. Seems like https://github.com/Automattic/jetpack/pull/18921 solves for this though.

**Comment by edequalsawesome** [comment]
> Show Links if the user has active links.

@sfougnier FWIW, when a new site is created on WordPress.com (either a simple site or a site that starts off AT'd with the eCommerce plan), it looks like we add links to the blogroll during the site creation process, so as of right now it looks like all sites will be shown to have active links.

I just created a simple site a few minutes ago, and these links were added by default:

[![Automatically Added Blogroll Links](https://d.pr/i/rUUWx2/H8gzgOqQWK+)](https://d.pr/i/rUUWx2/H8gzgOqQWK)

**Comment by getdave** [comment]
Given that 90% of users probably won't be making use of this feature I recommend **we remove the Links manager entirely**. As we cannot detect if the user is making use of the feature this is the only way forward.

Any users who need the screen can ask Happiness who can provide them with the direct link. 

This obeys the 80/20 rule.

Given how close we are to launch however, I don't see us being able to address this until post-launch. Possibly best to go out with Jetpack 9.6 in early April.

**Comment by davemart-in** [comment]
+1 to everything @getdave said. The links manager was [transitioned to a plugin](https://core.trac.wordpress.org/ticket/21307#comment:73) in Core 8 years ago.

**Comment by aleone89** [comment]
A user has asked here: https://wordpress.com/forums/topic/links-256/#post-3648616 where the links went from the menu. 

Just my two cents on this - if it's available in WP Admin, and the idea is to unify all experiences across the board - it should perhaps be available - or at best, both the new navigation and WP Admin view should look the same. 

I hope that's useful.

**Comment by kathrynwp** [comment]
Clarifying for HEs with users looking for the missing link manager that it's at 

`/wp-admin/link-manager.php`

**Comment by donalirl** [comment]
Anyone know how to access Link Categories now? I can't seem to track down the URL.
![image](https://user-images.githubusercontent.com/30727666/114803840-9356ac00-9de3-11eb-9548-c911f03a4efd.png)

**Comment by keoshi** [comment]
@donalirl It should be:

`/wp-admin/edit-tags.php?taxonomy=link_category`
